python/cudf/cudf/utils/_numba.py

```diff
@@ -133,18 +135,19 @@ def _setup_numba():
                 numba_config.CUDA_ENABLE_PYNVJITLINK = True
 
 
-class _CUDFNumbaConfig:
-    def __enter__(self):
-        self.CUDA_LOW_OCCUPANCY_WARNINGS = (
-            numba_config.CUDA_LOW_OCCUPANCY_WARNINGS
-        )
-        numba_config.CUDA_LOW_OCCUPANCY_WARNINGS = 0
+@contextlib.contextmanager
```

Comment:
Sometimes we choose to manually implement context managers because the overhead of `contextlib.contextmanager` is pretty high. Is this such a case? The history / blame might tell us.

Reply:
It appears this was added in https://github.com/rapidsai/cudf/pull/13337 without comment about performance, but locally I do see that the decorator version is almost 4x slower than the class version so I'll change this back to the class version.